clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3155,6 +3171,62 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
     KernelFunc->setInvalidDecl();
 }
 
+// For a wrapped parallel_for, copy attributes from original
+// kernel to wrapped kernel.
+void Sema::copyAttributes(const CXXRecordDecl *KernelObj) {
+  // Get the operator() function of the wrapper
+  CXXMethodDecl *OpParens = nullptr;
+  for (auto *MD : KernelObj->methods()) {
+    if (MD->getOverloadedOperator() == OO_Call) {
```

Comment:
What happens if there's more than one overloaded call operator? Or is that not possible because the `KernelObj` is somehow under our control?

Reply:
Yes, there will be only one operator() per kernel.

Reply:
Great, thanks for confirming!